strapi/strapi
# enhancement: add warning dialog when opening preview with unsaved changes

### What does it do?

Triggers a warning dialog when trying to open preview while the edit view has unsaved changes. If the user selects the save option, we save the document before redirecting.
### Why is it needed?

because of the designs, and because it prevents accidentally abandoning unsaved changes
### How to test it?

Open a content type that has preview enabled, make changes, don't save, open preview. You see the dialog. If you select save, it will save before redirecting. If you select don't save, it won't save before redirecting
